cms/release_calendar/viewsets.py

```diff
@@ -0,0 +1,51 @@
+from django.db.models import QuerySet
+from django.utils.translation import gettext_lazy as _
+from wagtail.admin.ui.tables import Column, DateColumn
+from wagtail.admin.views.chooser import PageStatusColumn
+from wagtail.admin.views.generic.chooser import ChooseResultsView, ChooseView
+from wagtail.admin.viewsets.chooser import ChooserViewSet
+
+from cms.release_calendar.enums import ReleaseStatus
+from cms.release_calendar.models import ReleaseCalendarPage
+
+
+class FutureReleaseCalendarMixin:
+    def get_object_list(self) -> QuerySet[ReleaseCalendarPage]:
+        return ReleaseCalendarPage.objects.exclude(status__in=[ReleaseStatus.CANCELLED, ReleaseStatus.PUBLISHED])
+
+    @property
+    def columns(self) -> list[Column]:
+        return [
+            self.title_column,  # type: ignore[attr-defined]
+            Column("release_date"),
+            Column("release_status", label=_("Status"), accessor="get_status_display"),
+            DateColumn(
+                "updated",
+                label=_("Last Updated"),
+                width="12%",
+                accessor="latest_revision_created_at",
+            ),
+            PageStatusColumn("status", label=_("Page status"), width="12%"),
```

Comment:
Non-blocker: I found seeing both Status and Page status in the chooser confusing. Is page status important enough to show, it doesn't impact the selection?. Though, that might just be me 🤷

Reply:
I didn't have it originally, but then I though it may be useful, added it. Will drop

Reply:
cc4da27